DOCS-11719 OneDrive Connector 1.0 (#1594)
mulesoft/docs-connectors: microsoft-onedrive/1.0/modules/ROOT/pages/microsoft-onedrive-connector-reference.adoc

> = Microsoft OneDrive Module Connector Reference

Reviewer: DriveItem in the descriptive text should really be "drive item," but it's up to you whether to expend the effort on this.

Author: It's actually called a driveItem in the Microsoft API documentation.

| Reconnection Strategy a| * <<Reconnect>>
* <<ReconnectForever>> |  A retry strategy in case of connectivity errors. |  | 
|===

=== Output

[%autowidth.spread]
|===
|Type |Binary
| Attributes Type a| Binary
|===

=== For Configurations

* <<OnedriveConfig>> 

=== Throws

* MICROSOFTONEDRIVE:BAD_REQUEST 
* MICROSOFTONEDRIVE:CONFLICT 
* MICROSOFTONEDRIVE:CONNECTIVITY 
* MICROSOFTONEDRIVE:FORBIDDEN 
* MICROSOFTONEDRIVE:INSUFFICIENT_STORAGE 
* MICROSOFTONEDRIVE:INVALID_CONNECTION 
* MICROSOFTONEDRIVE:INVALID_INPUT 
* MICROSOFTONEDRIVE:LOCKED 
* MICROSOFTONEDRIVE:NOT_FOUND 
* MICROSOFTONEDRIVE:RETRY_EXHAUSTED 
* MICROSOFTONEDRIVE:SERVER_ERROR 
* MICROSOFTONEDRIVE:TIMEOUT 
* MICROSOFTONEDRIVE:TOO_MANY_REQUESTS 
* MICROSOFTONEDRIVE:UNAUTHORIZED 

[[CopyDriveItem]]
== Copy Drive Item
`<microsoftOneDrive:copy-drive-item>`

Executes a copy request to Microsoft OneDrive.

=== Parameters

[%header%autowidth.spread]
|===
| Name | Type | Description | Default Value | Required
| Configuration | String | Name of the configuration to use. | | x
| DriveItem ID a| String | ID of the driveItem to copy. |  | x

> | Parent DriveItem ID a| String | ID of the folder to copy the selected driveItem to. If unspecified, the selected file is copied in the same folder in which it resides. |  | 

Reviewer: ```suggestion
| Parent DriveItem ID a| String | ID of the folder to copy the selected driveItem to. If unspecified, the selected file is copied to the same folder in which it resides. |  | 
```

> | Parent DriveItem ID a| String | ID of the folder to copy the selected driveItem to. If unspecified, the selected file is copied in the same folder in which it resides. |  | 

Reviewer: Maybe:

...is duplicated in the same folder in which it resides.

...is copied to the same folder in which it resides.

|===

=== Output

[%autowidth.spread]
|===
|Type |Binary
| Attributes Type a| Binary
|===

=== For Configurations

* <<OnedriveConfig>> 

=== Throws

* MICROSOFTONEDRIVE:BAD_REQUEST 
* MICROSOFTONEDRIVE:CONFLICT 
* MICROSOFTONEDRIVE:CONNECTIVITY 
* MICROSOFTONEDRIVE:FORBIDDEN 
* MICROSOFTONEDRIVE:INSUFFICIENT_STORAGE 
* MICROSOFTONEDRIVE:INVALID_CONNECTION 
* MICROSOFTONEDRIVE:INVALID_INPUT 
* MICROSOFTONEDRIVE:LOCKED 
* MICROSOFTONEDRIVE:NOT_FOUND 
* MICROSOFTONEDRIVE:RETRY_EXHAUSTED 
* MICROSOFTONEDRIVE:SERVER_ERROR 
* MICROSOFTONEDRIVE:TIMEOUT 
* MICROSOFTONEDRIVE:TOO_MANY_REQUESTS 
* MICROSOFTONEDRIVE:UNAUTHORIZED 

[[CreateDriveItem]]
== Create Drive Item
`<microsoftOneDrive:create-drive-item>`

Creates a new driveItem entity type in the specified OneDrive.

=== Parameters

[%header%autowidth.spread]
|===
| Name | Type | Description | Default Value | Required
| Configuration | String | The name of the configuration to use. | | x
| Parent DriveItem ID a| String |  ID of the folder to copy the selected driveItem to. |  | x
| Drive Item name a| String |  Name to use when creating the file. |  | x
| DriveItem Type a| Enumeration, one of:

** FOLDER

> ** FILE |  The type of driveItem to be created, it can be either FILE or FOLDER |  FILE | 

Reviewer: ```suggestion
** FILE |  The type of driveItem to create, which can be either FILE or FOLDER |  FILE | 
```

| Drive ID a| String |  Drive ID used to locate a DriveItem. |  | 
| Target Variable a| String |  Name of the variable that stores the operation's output. |  | 
| Target Value a| String |  Expression that evaluates the operation’s output. The outcome of the expression is stored in the *Target Variable* field. |  #[payload] | 
| Reconnection Strategy a| * <<Reconnect>>
* <<ReconnectForever>> |  A retry strategy in case of connectivity errors. |  | 
|===

=== Output

[%autowidth.spread]
|===
|Type |Binary
| Attributes Type a| Binary
|===

=== For Configurations

* <<OnedriveConfig>> 

=== Throws

* MICROSOFTONEDRIVE:BAD_REQUEST 
* MICROSOFTONEDRIVE:CONFLICT 
* MICROSOFTONEDRIVE:CONNECTIVITY 
* MICROSOFTONEDRIVE:FORBIDDEN 
* MICROSOFTONEDRIVE:INSUFFICIENT_STORAGE 
* MICROSOFTONEDRIVE:INVALID_CONNECTION 
* MICROSOFTONEDRIVE:INVALID_INPUT 
* MICROSOFTONEDRIVE:LOCKED 
* MICROSOFTONEDRIVE:NOT_FOUND 
* MICROSOFTONEDRIVE:RETRY_EXHAUSTED 
* MICROSOFTONEDRIVE:SERVER_ERROR 
* MICROSOFTONEDRIVE:TIMEOUT 
* MICROSOFTONEDRIVE:TOO_MANY_REQUESTS 
* MICROSOFTONEDRIVE:UNAUTHORIZED 

[[ListChildren]]
== List Children
`<microsoftOneDrive:list-children>`

> Retrieves a collection of DriveItems in the children relationship of a DriveItem.

Reviewer: Maybe "Retrieves a collection of DriveItems from the DriveItem's child relationships."

| Target Variable a| String |  Name of the variable that stores the operation's output. |  | 
| Target Value a| String |  Expression that evaluates the operation’s output. The outcome of the expression is stored in the *Target Variable* field. |  #[payload] | 
| Reconnection Strategy a| * <<Reconnect>>
* <<ReconnectForever>> |  A retry strategy in case of connectivity errors. |  | 
|===

=== Output

[%autowidth.spread]
|===
|Type |Any
|===

=== For Configurations

* <<OnedriveConfig>> 

=== Throws

* MICROSOFTONEDRIVE:BAD_REQUEST 
* MICROSOFTONEDRIVE:CONFLICT 
* MICROSOFTONEDRIVE:CONNECTIVITY 
* MICROSOFTONEDRIVE:FORBIDDEN 
* MICROSOFTONEDRIVE:INSUFFICIENT_STORAGE 
* MICROSOFTONEDRIVE:INVALID_CONNECTION 
* MICROSOFTONEDRIVE:INVALID_INPUT 
* MICROSOFTONEDRIVE:LOCKED 
* MICROSOFTONEDRIVE:NOT_FOUND 
* MICROSOFTONEDRIVE:RETRY_EXHAUSTED 
* MICROSOFTONEDRIVE:SERVER_ERROR 
* MICROSOFTONEDRIVE:TIMEOUT 
* MICROSOFTONEDRIVE:TOO_MANY_REQUESTS 
* MICROSOFTONEDRIVE:UNAUTHORIZED 

== Types
[[ProxyConfiguration]]
=== Proxy Configuration

[%header,cols="20s,25a,30a,15a,10a"]
|===
| Field | Type | Description | Default Value | Required

> | Host a| String | Host where the proxy requests is sent. |  | x

Reviewer: ```suggestion
| Host a| String | Host where the proxy requests are sent. |  | x
```

> | Port a| Number | Port where the proxy requests is sent. |  | x

Reviewer: ```suggestion
| Port a| Number | Port where the proxy requests are sent. |  | x
```
